airflow/contrib/operators/docker_swarm_operator.py

```diff
@@ -89,17 +90,22 @@ class DockerSwarmOperator(DockerOperator):
     :param tty: Allocate pseudo-TTY to the container of this service
         This needs to be set see logs of the Docker container / service.
     :type tty: bool
+    :param enable_logging: Show the application's logs in operator's logs.
+        Supported only if the Docker engine is using json-file or journald logging drivers.
+        The `tty` parameter should be set to use this with Python applications.
+    :type enable_logging: bool
     """
 
     @apply_defaults
     def __init__(
             self,
             image,
+            enable_logging=True,
             *args,
             **kwargs):
-
         super().__init__(image=image, *args, **kwargs)
 
+        self.enable_logging = enable_logging
         self.service = None
 
     def _run_image(self):
```

Comment:
Not really related to the PR but I found this
`types.Resources(mem_limit=self.mem_limit)` should better be `types.Resources(mem_limit=docker.utils.parse_bytes(self.mem_limit))` or we get non-parseable errors from the API (becomes float which is bad or string which is also bad)

Appears to work in your test though... :confused:

Reply:
The test never actually calls the API, so I think it won't catch this.

Reply:
Airflow admins would probably require a new ticket for this.